tests/testthat/test-shinytest2-teal.R

```diff
@@ -0,0 +1,368 @@
+simple_teal_data <- within(teal_data(), {
+  iris <- iris
+  mtcars <- mtcars
+})
+datanames(simple_teal_data) <- c("iris", "mtcars")
+
+report_module <- function(label = "example teal module") {
+  module(
+    label = label,
+    server = function(id, data, reporter) {
+      moduleServer(id, function(input, output, session) {
+        teal.reporter::simple_reporter_srv(
+          id = "reporter",
+          reporter = reporter,
+          card_fun = function(card) card
+        )
+        updateSelectInput(session, "dataname", choices = isolate(datanames(data())))
+        output$dataset <- renderPrint({
+          req(input$dataname)
+          data()[[input$dataname]]
+        })
+      })
+    },
+    ui = function(id) {
+      ns <- NS(id)
+      sidebarLayout(
+        sidebarPanel(
+          teal.reporter::simple_reporter_ui(ns("reporter")),
+          selectInput(ns("dataname"), "Choose a dataset", choices = NULL)
+        ),
+        mainPanel(verbatimTextOutput(ns("dataset")))
+      )
+    }
+  )
+}
+
+test_that("teal app initializes with no errors", {
+  app_title <- "Custom Teal App Title"
+  app_favicon <- "https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/PNG/teal.png"
+  app_header <- "Custom Teal App Header"
+  app_footer <- "Custom Teal App Footer"
+  app <- get_test_app_object(
+    data = simple_teal_data,
+    modules = example_module(label = "Example Module"),
+    title = build_app_title(
+      app_title,
+      app_favicon
+    ),
+    header = app_header,
+    footer = app_footer
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+  expect_no_shiny_error(app)
+
+  expect_equal(
+    app$get_html("head > title")[1] |>
+      rvest::read_html() |>
+      rvest::html_text(),
+    app_title
+  )
+  expect_equal(
+    app$get_html("head > link[rel='icon']") |>
+      rvest::read_html() |>
+      rvest::html_elements("link") |>
+      rvest::html_attr("href"),
+    app_favicon
+  )
+  expect_true(
+    grepl(
+      app_header,
+      app$get_html("header") |>
+        rvest::read_html() |>
+        rvest::html_text()
+    )
+  )
+  expect_true(
+    grepl(
+      app_footer,
+      app$get_html("footer") |>
+        rvest::read_html() |>
+        rvest::html_text()
+    )
+  )
```

Comment:
Given the title "initializes with no errors" tests should stop on `expect_no_shiny_error(app)`
```suggestion
```